### PR TITLE
Fix dryRun issue and avoid unnecessary log msg when node affinity strategy is disabled.

### PR DESCRIPTION
**Pull request by aveshagarwal**
@ravisantoshgudimetla @ingvagabund

### CODE REVIEW

File: pkg/descheduler/strategies/node_affinity.go [hunk 1]
```diff
@@ -70,5 +69,6 @@ func removePodsViolatingNodeAffinityCount(ds *options.DeschedulerServer, strateg
 			return evictedPodCount
 		}
 	}
+	glog.V(1).Infof("Evicted %v pods", evictedPodCount)
```
Comment (ingvagabund) [inline pkg/descheduler/strategies/node_affinity.go]: How does it help to move the log message here? It is logged in both cases under the same conditions.

Comment (aveshagarwal) [reply pkg/descheduler/strategies/node_affinity.go]: previously this msg was being logged even when this strategy was disabled so want to avoid it.

Comment (ingvagabund) [reply pkg/descheduler/strategies/node_affinity.go]: I see now:
```
 	if !strategy.Enabled {
 		return evictedPodCount
 	}
```

**Comment by aveshagarwal** [comment]
@ingvagabund thanks.

**Pull request merged**